browser/content_settings/brave_content_settings_registry_unittest.cc

```diff
@@ -0,0 +1,185 @@
+      ContentSettingsType::BRAVE_TRACKERS,
+      ContentSettingsType::BRAVE_HTTP_UPGRADABLE_RESOURCES,
+      ContentSettingsType::BRAVE_FINGERPRINTING_V2,
+      ContentSettingsType::BRAVE_SHIELDS,
+      ContentSettingsType::BRAVE_REFERRERS,
+      ContentSettingsType::BRAVE_COOKIES,
+      ContentSettingsType::BRAVE_SPEEDREADER,
+  };
+
+  for (const ContentSettingsInfo* info : *registry()) {
+    SCOPED_TRACE("Content setting: " + info->website_settings_info()->name());
+
+    if (info->website_settings_info()->type() <
+        ContentSettingsType::BRAVE_START)
+      continue;
+
+    if (info->incognito_behavior() ==
+        ContentSettingsInfo::INHERIT_IN_INCOGNITO) {
+      EXPECT_THAT(safe_types, Contains(info->website_settings_info()->type()));
+    }
+  }
+}
+
+// Check the correct factory default setting is retrieved. Note the factory
+// default settings are hard coded, so changing them in ContentSettingsRegistry
+// would require this test to be updated.
+TEST_F(BraveContentSettingsRegistryTest, GetInitialDefaultSetting) {
+  const ContentSettingsInfo* info = nullptr;
+  {
+    SCOPED_TRACE("Content setting: BRAVE_ADS");
+    info = registry()->Get(ContentSettingsType::BRAVE_ADS);
+    EXPECT_EQ(CONTENT_SETTING_BLOCK, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_TRACKERS");
+    info = registry()->Get(ContentSettingsType::BRAVE_TRACKERS);
+    EXPECT_EQ(CONTENT_SETTING_BLOCK, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_HTTP_UPGRADABLE_RESOURCES");
+    info =
+        registry()->Get(ContentSettingsType::BRAVE_HTTP_UPGRADABLE_RESOURCES);
+    EXPECT_EQ(CONTENT_SETTING_BLOCK, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_SHIELDS");
+    info = registry()->Get(ContentSettingsType::BRAVE_SHIELDS);
+    EXPECT_EQ(CONTENT_SETTING_ALLOW, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_REFERRERS");
+    info = registry()->Get(ContentSettingsType::BRAVE_REFERRERS);
+    EXPECT_EQ(CONTENT_SETTING_BLOCK, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_SPEEDREADER");
+    info = registry()->Get(ContentSettingsType::BRAVE_SPEEDREADER);
+    EXPECT_EQ(CONTENT_SETTING_ALLOW, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_ETHEREUM");
+    info = registry()->Get(ContentSettingsType::BRAVE_ETHEREUM);
+    EXPECT_EQ(CONTENT_SETTING_ASK, info->GetInitialDefaultSetting());
+  }
+
+  {
+    SCOPED_TRACE("Content setting: BRAVE_SOLANA");
+    info = registry()->Get(ContentSettingsType::BRAVE_SOLANA);
+    EXPECT_EQ(CONTENT_SETTING_ASK, info->GetInitialDefaultSetting());
+  }
+
+  // TODO(bridiver) - DEFAULT is not a valid setting, but leaving it for
+  // now because this requires more extensive changes to fix correctly
+  // {
+  //   SCOPED_TRACE("Content setting: BRAVE_FINGERPRINTING_V2");
```

Comment:
i'd rather keep checks that assert "not valid" things then keep something commented and not even compiled

Reply:
uncommenting these causes a crash because GetInitialDefaultSetting DCHECKs for invalid values